exercises/practice/protein-translation/runtests.jl

```diff
@@ -0,0 +1,88 @@
+using Test
+
+include("protein-translation.jl")
+
+@testset "Protein Translation" begin
+
+    @testset "Empty RNA sequence returns an empty list" begin
+        @test rna_to_amino_acids("") == []
+    end
+
+    @testset "Methionine RNA sequence is decoded as Methionine" begin
+        @test rna_to_amino_acids("AUG") == ["Methionine"]
+    end
+
+    @testset "Phenylalanine RNA sequence is decoded as Phenylalanine" begin
+        @test rna_to_amino_acids("UUUUUC") == ["Phenylalanine", "Phenylalanine"]
+    end
+
+    @testset "Leucine RNA sequence is decoded as Leucine" begin
+        @test rna_to_amino_acids("UUA") == ["Leucine"]
+    end
+
+    @testset "Leucine RNA sequence is decoded as Leucine" begin
+        @test rna_to_amino_acids("UUG") == ["Leucine"]
+    end
+
+    @testset "Serine RNA sequence is decoded as Serine" begin
+        @test rna_to_amino_acids("UCUUCCUCAUCG") == ["Serine", "Serine", "Serine", "Serine"]
+    end
+
+    @testset "Tyrosine RNA sequence is decoded as Tyrosine" begin
+        @test rna_to_amino_acids("UAUUAC") == ["Tyrosine", "Tyrosine"]
+    end
+
+    @testset "Cysteine RNA sequence is decoded as Cysteine" begin
+        @test rna_to_amino_acids("UGUUGC") == ["Cysteine", "Cysteine"]
+    end
+
+    @testset "Tryptophan RNA sequence is decoded as Tryptophan" begin
+        @test rna_to_amino_acids("UGG") == ["Tryptophan"]
+    end
+
+    @testset "STOP codon terminates translation" begin
+        @test rna_to_amino_acids("UAA") == []
+        @test rna_to_amino_acids("UAG") == []
+        @test rna_to_amino_acids("UGA") == []
+    end
+
+    @testset "Sequence of two codons translates into proteins" begin
+        @test rna_to_amino_acids("UUUUUUUGA") == ["Phenylalanine", "Phenylalanine"]
+    end
+
+    @testset "Sequence of two different codons translates into proteins" begin
+        @test rna_to_amino_acids("UUAUUUUAG") == ["Leucine", "Phenylalanine"]
+    end
+
+    @testset "Translation stops if STOP codon appears in middle of sequence" begin
+        @test rna_to_amino_acids("UGGUAAUGCAUG") == ["Tryptophan"]
+    end
+
+    @testset "Translation stops if STOP codon appears at beginning of sequence" begin
+        @test rna_to_amino_acids("UAGUAUUCGUCAUCU") == []
+    end
+
+    @testset "Translation stops if STOP codon appears at end of two-codon sequence" begin
+        @test rna_to_amino_acids("UGGUGUUGA") == ["Tryptophan", "Cysteine"]
+    end
+
+    @testset "Non existent codon causes translation exception" begin
+        @test_throws TranslationError rna_to_amino_acids("AAA")
+    end
+
+    @testset "Incomplete codon causes translation exception" begin
+        @test_throws TranslationError rna_to_amino_acids("UGUU")
+    end
```

Comment:
Requiring the user to define a custom exception type makes this exercise less approachable for beginners and may feel like make-work to more experienced programmers.

The norm in Julia is to throw an ArgumentError or a DomainError:

```
julia> parse(Int, "lmao")
ERROR: ArgumentError: invalid base 10 digit 'l' in "lmao"
```

My preference is that we take one of these two options:

1: require user to throw an argumenterror (will require some minor changes to example.jl)

```suggestion
    @testset "Non existent codon causes translation exception" begin
        @test_throws ArgumentError rna_to_amino_acids("AAA")
    end

    @testset "Incomplete codon causes translation exception" begin
        @test_throws ArgumentError rna_to_amino_acids("UGUU")
    end
```

2: require them to throw some kind of exception, but don't worry about the details:

```suggestion
    @testset "Non existent codon causes translation exception" begin
        @test_throws Exception rna_to_amino_acids("AAA")
    end

    @testset "Incomplete codon causes translation exception" begin
        @test_throws Exception rna_to_amino_acids("UGUU")
    end
```

I don't have a preference between those two options.

The other option is that we explain in the instructions.append.md how and why you should define your own exception type. I can't do that because I don't believe that it's worthwhile to define your own exception types :D